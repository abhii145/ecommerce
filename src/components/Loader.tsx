import React, { memo } from 'react';

const Loader = () => {
  return (
    <div className="flex justify-center items-center h-screen" role="status" aria-label="Loading">
      <div className="border-t-4 border-blue-500 border-solid w-16 h-16 rounded-full animate-spin"></div>
      <span className="sr-only">Loading...</span>
    </div>
  );
}

export default memo(Loader);
