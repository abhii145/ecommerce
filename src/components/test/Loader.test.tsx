import React from "react"
import { render } from "@testing-library/react"
import Loader from "../Loader"

describe("Loader component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Loader />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Loader component", () => {
    const { container } = render(<Loader />)
    expect(container.firstChild).toHaveClass(
      "flex justify-center items-center h-screen"
    )
  })
})
