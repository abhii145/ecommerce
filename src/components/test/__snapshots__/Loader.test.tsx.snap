// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading"
    class="flex justify-center items-center h-screen"
    role="status"
  >
    <div
      class="border-t-4 border-blue-500 border-solid w-16 h-16 rounded-full animate-spin"
    />
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
